### PR TITLE
DevDocs: New CardHeading component

### PR DESCRIPTION
**Pull request by scruffian**
This is another take on #23732.

The idea is to share styles for our Card headings so that we cut down on duplicated CSS and also have more consistency in our headings. This will encourage the use of standardized spacing and type sizes.

Here it is in DevDocs:
screen shot 2018-04-04 at 15 09 20

Here it is in the wild:
screen shot 2018-04-04 at 15 10 15

### CODE REVIEW

**Comment by shaunandrews** [comment]
Hmmm. I'm not sure I understand how/why this is better than just using standard heading tags and defining stylings in `components/card/style`?

**Comment by shaunandrews** [comment]
> Where it really gets useful is when you add options for the card title. Mineral UI provides a lot of options for their card title component.

You've sold me.

**Comment by scruffian** [comment]
> Hmmm. I'm not sure I understand how/why this is better than just using standard heading tags and defining stylings in components/card/style?

One big issue with this is that we would tie the semantics of the title to the style, which is something we should avoid.

**Comment by drw158** [comment]
2 things for later:

- I'm wondering if there should be a default tag and size if none is provided. I think this would make it easier to use. Sort of like how we have a default Gridicon size if none is provided.
- Could this be used outside of a Card? If so, we should probably say something like, "This component should only be used in the Card component"

File: client/components/card-heading/index.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+function CardHeading( { tagName = 'h1', size = 21, children } ) {
+	const classNameObject = {};
+	classNameObject[ 'card-heading-' + size ] = includes( validTypeSizes, size );
+	const classes = classNames( 'card-heading', classNameObject );
+	return React.createElement( tagName, { className: classes }, preventWidows( children, 2 ) );
+}
+
+CardHeading.propTypes = {
+	tagName: PropTypes.string,
```
Comment (ryelle) [inline client/components/card-heading/index.jsx]: I know this is already merged, but what about using `PropTypes.oneOf( [ 'h1', 'h2', 'h3', 'h4', 'h4', 'h6' ] )` to make sure the title is using semantic tags?

Comment (scruffian) [reply client/components/card-heading/index.jsx]: Great idea. Fancy creating a PR?

**Comment by scruffian** [comment]
>I'm wondering if there should be a default tag and size if none is provided. I think this would make it easier to use. Sort of like how we have a default Gridicon size if none is provided. This way, I can just used the component without digging into the documentation, and I know it'll use a common size text.

There is. It's H1 and 24px.

Could this be used outside of a Card? If so, we should probably say something like, "This component should only be used in the Card component"

Good idea 👍

**Pull request merged**